### PR TITLE
Support for changing crypto type and FICON type of adapters; Removed --crypto-number

### PR DESCRIPTION
**Pull request by andy-maier**
For details, see the commit message.

Note, this requires zhmcclient PR https://github.com/zhmcclient/python-zhmcclient/pull/1177, which is targeted for zhmcclient 1.8.0.

### CODE REVIEW

File: zhmccli/_cmd_adapter.py [hunk 1]
```diff
@@ -28,6 +31,31 @@
...
 
+# Mapping of --crypto-type option values to crypto-type adapter property values
+CRYPTO_TYPE_PROPERTIES = {
+    'acc': 'accelerator',
+    'cca': 'cca-coprocessor',
+    'ep11': 'ep11-coprocessor',
+}
+
+# List of --crypto-type option values
+CRYPTO_TYPE_OPTIONS = list(CRYPTO_TYPE_PROPERTIES.keys())
```
Comment (EdGue42) [inline zhmccli/_cmd_adapter.py:42]: are they sorted by default? should be.

File: zhmccli/_cmd_adapter.py [hunk 2]
```diff
@@ -119,19 +147,26 @@ def adapter_show(cmd_ctx, cpc, adapter):
...
 @click.option('--crypto-tke/--no-crypto-tke', default=None, required=False,
-              help='Permit TKE commands on the crypto adapter. '
-              '(Crypto only).')
+              help='The new setting whether TKE commands are permitted on the '
+              'crypto adapter. (Crypto only).')
+@click.option('--crypto-type', default=None, required=False,
+              type=click.Choice(CRYPTO_TYPE_OPTIONS),
+              help='The new crypto type (Crypto only; Adapter must be '
+              'offline).')
+@click.option('--ficon-type', default=None, required=False,
```
Comment (simon-spinner) [inline zhmccli/_cmd_adapter.py:163]: I would prefer --storage-type or --storage-protocol (used in the WSAPI book). --ficon-type=fcp looks weird.

File: zhmccli/_cmd_adapter.py (hunk 1 above)
Comment (andy-maier) [reply zhmccli/_cmd_adapter.py:42]: I would prefer to keep the order defined in the dictionary, but that does not work below Python 3.6 (unless we change to OrderedDict).
 
 Changed this to sort the list, and also for FICON_TYPE_OPTIONS.

File: zhmccli/_cmd_adapter.py (hunk 2 above)
Comment (andy-maier) [reply zhmccli/_cmd_adapter.py]: The physical adapter card is named "FICON Express", and the input parameter of the "Change Adapter Type" operation is named "type". Hence I came up with the name "ficon-type". The corresponding property on the Storage Group object is also named "type".

I have changed it to "storage-type".

**Pull request merged**